fedora_messaging/tests/unit/test_cli.py

```diff
@@ -369,3 +374,135 @@ def test_consume_halt_without_exitcode(self, mock_consume, mock_importlib):
         mock_importlib.import_module.called_once_with("mod")
         mock_consume.assert_called_once_with(mock_mod_with_callable.callable, "b")
         self.assertEqual(0, result.exit_code)
+
+
+class PublishCliTests(unittest.TestCase):
+    """Unit tests for the 'publish' command of the CLI."""
+
+    def setUp(self):
+        self.runner = CliRunner()
+
+    @mock.patch("fedora_messaging.cli.api.publish")
+    def test_correct_msg_in_file(self, mock_publish):
+        """Assert providing path to file with correct message via the CLI works."""
+        cli_options = {"file": GOOD_MSG_DUMP, "exchange": "test_pe"}
+        result = self.runner.invoke(
+            cli.cli,
+            [
+                "--conf=" + GOOD_CONF,
+                "publish",
+                "--exchange=" + cli_options["exchange"],
+                cli_options["file"],
+            ],
+        )
+        self.assertIn("Sending message with topic test_topic", result.output)
+        mock_publish.assert_called_once()
+        self.assertEqual(mock_publish.call_args_list[0][0][0].topic, "test_topic")
```

Comment:
It'd be clearer here to assign ``mock_publish.call_args_list[0][0][0]`` to a variable like ``sent_msg`` or something. You should also be able to create an ``expected_msg`` object with the expected body, topic, etc. and just do ``self.assertEqual(expected_msg, sent_msg)`` since the message class implements the equality check.

Reply:
True, although you could set the ``expected_msg._headers["sent-at"]`` to be the same as the fixture message